### PR TITLE
✨ Option to verify branch

### PR DESCRIPTION
**Pull request by laurentsimon**
The default branch is set to `main`, and can be overwritten via `--branch xxx`. We check that the branch in the provenance information of the form `refs/heads/xxx`

### CODE REVIEW

File: pkg/provenance.go [hunk 1]
```diff
@@ -393,8 +395,148 @@ func VerifyProvenance(env *dsselib.Envelope, expectedHash string) error {
...
+	if version != 1 {
+		return "", fmt.Errorf("%w", errorInvalidVersion)
+	}
+
+	// RefType.
+	refType, err := getAsString(parameters, "ref_type")
+	if err != nil {
+		return "", err
+	}
+	// RefType.
```
Comment (asraa) [inline pkg/provenance.go:522]: nit: fix comment

File: pkg/provenance.go [hunk 2]
```diff
@@ -393,8 +395,148 @@ func VerifyProvenance(env *dsselib.Envelope, expectedHash string) error {
...
+	}
+
+	switch refType {
+	case "branch":
+		return ref, nil
+	case "tag":
+		return baseRef, nil
+	}
+
+	return "", fmt.Errorf("%w: %s", errorInvalidDssePayload,
```
Comment (asraa) [inline pkg/provenance.go:540]: can you combine this?
```
fmt.Errorf("%w: %s %s", errorInvalidDssePayload, "unknown ref type", refType)
```

**Review by asraa — CHANGES_REQUESTED** [review]

File: pkg/provenance.go (hunk 2 above)
Comment (laurentsimon) [reply pkg/provenance.go]: done.

File: pkg/provenance.go [hunk 3]
```diff
@@ -393,8 +395,137 @@ func VerifyProvenance(env *dsselib.Envelope, expectedHash string) error {
...
+	if !ok {
+		return "", fmt.Errorf("%w: %s", errorInvalidDssePayload, "parameters type payload")
+	}
+
+	bRef, ok := payload["base_ref"]
+	if !ok {
+		return "", fmt.Errorf("%w: %s", errorInvalidDssePayload, "parameters type base_ref")
+	}
+
+	baseRef, ok = bRef.(string)
```
Comment (asraa) [inline pkg/provenance.go:483]: can you `getAsString` the baseRef?

Comment (laurentsimon) [reply pkg/provenance.go:483]: that does not work. getAsString needs a map, the input is just a string. Right?

Comment (asraa) [reply pkg/provenance.go:483]: getAsString(payload, 'base_ref')?

Comment (laurentsimon) [reply pkg/provenance.go:483]: done, thanks!

**Review by asraa — APPROVED** [review]

**Pull request merged**